### PR TITLE
Abcl/1.8.0/rc/10

### PR DESCRIPTION
**Pull request by easye**
Supersedes #334

### CODE REVIEW

File: src/org/armedbear/lisp/JarPathname.java [hunk 1]
```diff
@@ -397,7 +397,7 @@ public boolean isLocalFile() {
     if (p.isLocalFile()) {
```
Comment (rec) [inline src/org/armedbear/lisp/JarPathname.java:427]: You could collapse these four lines into:

    return p.isLocalFile();

File: src/org/armedbear/lisp/ZipCache.java [hunk 2]
```diff
@@ -423,18 +422,79 @@ synchronized public static Archive getArchive(JarPathname jar) {
     if (innerJars.equals(NIL)) {
       return getArchiveFile(jar);
     } else {
```
Comment (rec) [inline src/org/armedbear/lisp/ZipCache.java:424]: An `else` after an early `return` is an anti-pattern!

It forces a needless extra level of indentation.

File: src/org/armedbear/lisp/ZipCache.java [hunk 3]
```diff
@@ -423,18 +422,79 @@ synchronized public static Archive getArchive(JarPathname jar) {
...
+        JarPathname nextAsEntry = new JarPathname();
+        nextAsEntry
+          .setDevice(nextArchive.getDevice())
+          .setDirectory(nextJarArchive.getDirectory())
+          .setName(nextJarArchive.getName())
+          .setType(nextJarArchive.getType());
+        // FIXME 
+        // The pathnames for subsquent entries in a PATHNAME-JAR are relative.  Should they be?
+        if (nextAsEntry.getDirectory().car().equals(Keyword.RELATIVE)) {
+          LispObject directories = nextAsEntry.getDirectory();
```
Comment (rec) [inline src/org/armedbear/lisp/ZipCache.java:449]: One whole line for the `.cdr` is sprawly.  Why not 

    LispObject directories = nextAsEntry.getDirectory().cdr();

?

File: src/org/armedbear/lisp/ZipCache.java [hunk 4]
```diff
@@ -423,18 +422,79 @@ synchronized public static Archive getArchive(JarPathname jar) {
...
+        }
+        stream = new ArchiveStream(inputStream, nextArchive, entry);
+        result = stream;
+        cache.put(nextArchive, result); 
+
+        innerJars = innerJars.cdr();
+        if (innerJars.cdr().equals(NIL)
+            && (!jar.getDirectory().equals(NIL)
+                || !jar.getName().equals(NIL)
+                || !jar.getType().equals(NIL))) {
```
Comment (rec) [inline src/org/armedbear/lisp/ZipCache.java:474]: if (innerJars.cdr().equals(NIL)
            && !(jar.getDirectory().equals(NIL)
                && jar.getName().equals(NIL)
                && jar.getType().equals(NIL)))

This statement is logically equivalent but has two fewer operations (and so is a bit smaller, clearer and faster).

**Review by rec — APPROVED** [review]
Random thoughts about a random piece of code I clicked on.  :-D

File: src/org/armedbear/lisp/ZipCache.java (hunk 4 above)
Comment (easye) [reply src/org/armedbear/lisp/ZipCache.java:474]: Right you are.

**Comment by easye** [comment]
Superseded by #336

**Pull request closed without merging**